src/ceph-volume/ceph_volume/util/prepare.py

```diff
@@ -458,9 +459,23 @@ def osd_mkfs_bluestore(osd_id, fsid, keyring=None, wal=False, db=False):
 
     command = base_command + supplementary_command
 
-    _, _, returncode = process.call(command, stdin=keyring, show_command=True)
-    if returncode != 0:
-        raise RuntimeError('Command failed with exit code %s: %s' % (returncode, ' '.join(command)))
+    """
+    When running in containers the --mkfs on raw device sometimes fails
+    to acquire a lock through flock() on the device because systemd-udevd holds one temporarily.
+    See KernelDevice.cc and _lock() to understand how ceph-osd acquires the lock.
+    Because this is really transient, we retry up to 5 times and wait for 1 sec in-between
+    """
+    for retry in range(5):
+        _, stderr, returncode = process.call(command, stdin=keyring, terminal_verbose=True, show_command=True)
+        if returncode != 0:
+            if any("(11) Resource temporarily unavailable" in err for err in stderr):
```

Comment:
Shouldn't we just check the return code here? We get ´-errno´ iiuc. https://github.com/ceph/ceph/blob/420a61c4034785757f8765c8b14e2b03185ad7f2/src/blk/kernel/KernelDevice.cc#L91

```suggestion
            if errno.EWOULDBLOCK:
```

With the appropriate import of course.

Reply:
ping @leseb

Reply:
I'm not sure, c-v was printing a different rc last time I checked, which was not `EWOULDBLOCK`. I think that's why I took that route, but because I was trying to handle the return code from Rook earlier, I got carried away. Thanks!

Reply:
Yeah c-v rewrites those unfortunately.